src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/StorePal.cs

```diff
@@ -25,12 +30,123 @@ public static IStorePal FromCertificate(ICertificatePal cert)
 
         public static IStorePal LinkFromCertificateCollection(X509Certificate2Collection certificates)
         {
-            throw new NotImplementedException();
+            return new OpenSslX509StoreProvider(certificates);
         }
 
         public static IStorePal FromSystemStore(string storeName, StoreLocation storeLocation, OpenFlags openFlags)
         {
+            if (storeLocation != StoreLocation.LocalMachine)
+            {
+                throw new NotImplementedException();
```

Comment:
Coming up with a plan for CurrentUser\My is on my radar ("RSA private key management"), and that should generalize to the rest of CU.  Other LocalMachine things might be... trickier :smile:.

CU\My is where support for ReadWrite comes in.  So all of these NotImplementedExceptions are expected to be functional or a different exception by RTM (I'll open issues for them).